src/cmd/sign.js

```diff
@@ -63,43 +62,20 @@ export default function sign(
       getIdFromFile(savedIdPath),
     ]);
 
-    const manifestId = getManifestId(manifestData);
+    const id = getManifestId(manifestData);
 
-    if (id && !manifestId) {
+    if (!id) {
       throw new UsageError(
-        `Cannot set custom ID ${id} - The add-on ID must be specified in the manifest.json file.`,
+        'The add-on ID must be specified in the manifest.json file.',
       );
     }
-    if (idFromSourceDir && !manifestId) {
+
+    if (idFromSourceDir && !id) {
```

Comment:
This code is now unreachable because you are checking and throwing before when `!id`.

Reply:
OK I figured out what was going on. PR updated.